### PR TITLE
Refactor routes to use pre-handler function for validation

### PR DESCRIPTION
**Pull request by Beckyrose200**
This PR refactors the existing route handlers to utilize a pre-handler function for validation. The pre-handler function will handle the validation logic, allowing for cleaner and more maintainable route handlers.

### CODE REVIEW

**Review by thousand-leaves — DISMISSED** [review]

File: src/routes/pre-handlers.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    if (validation.error) {
+      return h.redirect('/search-sbi').takeover()
+    }
+
+    request.params.sbi = validation.value.sbi
+
+    return h.continue
+  }
+}
+
```
Comment (natseg) [inline src/routes/pre-handlers.js:19]: **Recommended — async inconsistency**

`validateSbi` is synchronous but `validateCrn` is declared `async` despite containing no `await`. For consistency, either remove `async` from `validateCrn` or add it to `validateSbi`.

File: test/unit/routes/pre-handlers.test.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    let request
+
+    beforeEach(() => {
+      request = {
+        params: { sbi: '123456789' }
+      }
+    })
+
+    describe('when SBI is valid', () => {
+      beforeEach(() => {
```
Comment (natseg) [inline test/unit/routes/pre-handlers.test.js:50]: **Recommended — missing edge-case test**

The pre-handler uses `request.params?.sbi ?? ''` but no test exercises the case where `request.params` is `undefined` or missing the key. A single test confirming the fallback to empty string (and subsequent redirect) would improve confidence in that defensive code path.

File: test/unit/routes/pre-handlers.test.js [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        expect(result).toBe(h.continue)
+        expect(h.redirect).not.toHaveBeenCalled()
+      })
+
+      test('it should accept a valid numeric SBI', async () => {
+        request.params.sbi = '987654321'
+        const result = await validateSbi.method(request, h)
+
+        expect(result).toBe(h.continue)
+      })
```
Comment (natseg) [inline test/unit/routes/pre-handlers.test.js]: **Nit — redundant test assertion**

"it should accept a valid numeric SBI" duplicates the behaviour already covered by "it should allow the request to continue" since the mock setup is identical. Consider either consolidating these or giving distinct inputs to add genuine value.

File: src/routes/pre-handlers.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+export const validateSbi = {
+  method: (request, h) => {
+    const sbiInput = request.params?.sbi ?? ''
+    const validation = schemas.business.sbi.validate({ sbi: sbiInput })
+
+    if (validation.error) {
+      return h.redirect('/search-sbi').takeover()
+    }
+
```
Comment (natseg) [inline src/routes/pre-handlers.js:11]: **Nit — params mutation**

Mutating `request.params.sbi` to the Joi-coerced value is a good idea (ensures downstream handlers get the normalised value), but a brief inline comment explaining the intent would help future maintainers understand why mutation occurs in a pre-handler.

File: src/routes/pre-handlers.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import { schemas } from '@defra/fcp-sfd-frontend-engine'
```
Comment (natseg) [inline src/routes/pre-handlers.js:1]: **Nit — file location**

Happy with this living here for now. If additional pre-handlers accumulate (auth checks, pagination validation, etc.), consider splitting into a `src/routes/pre-handlers/` directory.

**Review by natseg — DISMISSED** [review]
Good refactoring — clean, idiomatic use of Hapi pre-handlers. A few minor suggestions below.

File: src/routes/pre-handlers.js (hunk 5 above)
Comment (Beckyrose200) [reply src/routes/pre-handlers.js:1]: Yeah good shout

**Review by natseg — APPROVED** [review]
